components/UnifiedDocFeed/utils/getSelectedUrlFilters.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  feedTypeOpts,
+  sortOpts,
+  tagFilters,
+  topLevelFilters,
+} from "../constants/UnifiedDocFilters";
+import { scopeOptions } from "~/config/utils/options";
+
+export type SelectedUrlFilters = {
+  topLevel: string | undefined;
+  type: string | undefined;
+  sort: string | undefined;
+  time: string | undefined;
+  tags: string[];
+};
+
+export const getSelectedUrlFilters = ({
+  query,
+  pathname,
+}): SelectedUrlFilters => {
+  const defaults = {
+    topLevel: topLevelFilters[0].value,
+    type: Object.values(feedTypeOpts)[0].value,
+    sort: sortOpts[0].value,
+    time: scopeOptions[0].value,
+    tags: <string[]>[],
+  };
+  const selected = { ...defaults };
+
+  if (!(query || pathname)) {
+    return defaults;
+  }
```

Comment:
This will prevent false positives on falsy JS values
```suggestion
  if (!(isEmpty(query ?? pathname)) {
    return defaults;
  }
```

Reply:
I actually want the `||` opeartor here. I want `""` to be falsy.

Though, LMK if you see some way which the above will fail

Reply:
in case query = 0 & pathname  = ""
- !(query || pathname) => !(false || false) => evaluates to true 
- isEmpty(query ?? pathname) =>  !isEmpty(0) => evalues to false

Above can lead to weird bugs. Former requires URL to be always be presented as intended. Latter disregards any edgecase